encrypted-media/scripts/playback-persistent-license-events.js

```diff
@@ -0,0 +1,167 @@
+            assert_equals(event.target, _mediaKeySession );
+            assert_true(event instanceof window.Event );
+            assert_equals(event.type, 'keystatuseschange' );
+
+            var hasKeys = false, pendingKeys = false;
+            _mediaKeySession.keyStatuses.forEach( function( value, keyid ) {
+                assert_any( assert_equals, value, [ 'status-pending', 'usable' ] );
+
+                hasKeys = true;
+                pendingKeys = pendingKeys || ( value === 'status-pending' );
+            });
+
+            if ( !_receivedAllKeysUsableEvent && hasKeys && !pendingKeys ) {
+                _receivedAllKeysUsableEvent = true;
+                _events.push( 'allkeysusable' );
+            }
+
+            if ( !hasKeys ) {
+                _events.push( 'emptykeyslist' );
+            }
+        }
+
+        function onEncrypted(event) {
+            assert_equals(event.target, _video);
+            assert_true(event instanceof window.MediaEncryptedEvent);
+            assert_equals(event.type, 'encrypted');
+
+            waitForEventAndRunStep('message', _mediaKeySession, onMessage, test);
+            _mediaKeySession.generateRequest(   config.initDataType || event.initDataType,
+                                                config.initData || event.initData ).then( function() {
+
+                _events.push( 'generaterequest' );
+                _sessionId = _mediaKeySession.sessionId;
+            }).catch(function(error) {
+                forceTestFailureFromPromise(test, error);
+            });
+        }
+
+        function onClosed(event) {
+            _events.push( 'closed-promise' );
+
+            setTimeout( test.step_func( function() {
+                assert_array_equals( _events,
+                                    [
+                                        'generaterequest',
+                                        'license-request',
+                                        'license-response',
+                                        'updated',
+                                        'allkeysusable',
+                                        'playing',
+                                        'remove',
+                                        'emptykeyslist',
+                                        'license-release',
+                                        'release-response',
+                                        'closed-promise',
+                                        'updated'
+                                    ],
+                                    "Expected events sequence" );
+
+                _video.setMediaKeys( null ).then( function() { test.done() } );
+            } ), 0 );
+        }
+
+        function onTimeupdate(event) {
+            if ( _video.currentTime > ( config.duration || 5 ) && !_receivedTimeupdateEvent ) {
+                _receivedTimeupdateEvent = true;
+                _video.pause();
+                _video.src = "";
```

Comment:
Global comment for this PR and probably all our files:
`.src = ""` is incorrect as it doesn't do what one might think it does. Use ` video.removeAttribute('src');` instead.
Also, to be sure you've freed the media player, you must follow it with `video.load();`